.github/workflows/pre-commit-check.yaml

```
@@ -0,0 +1,22 @@
+---
+name: pre-commit-check
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  build:
+    runs-on: macOS-latest
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Install prerequisites
+      run: |
+        brew install tfenv tflint terraform-docs pre-commit
+        pre-commit install
+        tfenv install
```

I can't get this to work without a `.terraform-version` file

---

yeah, it worked when I had the newer versions installed but other stuff broke for me. Do you have any info on how to use the latest version of pre-commit-terraform and pre-commit-hooks? I think the latter was crashing hard due to python2.7/3.0 issues.

---

I'd need to test it but we can install them all without brew, I haven't delved too deeply into how to create good GH Actions. Hashicorp did release their own GH action that installs terraform and a wrapper script - https://github.com/hashicorp/setup-terraform

Example installing all not using a package manager:
```sh
# tfenv
git clone https://github.com/tfutils/tfenv.git ~/.tfenv
ln -s ~/.tfenv/bin/* /usr/local/bin
# tflint
curl https://raw.githubusercontent.com/terraform-linters/tflint/master/install_linux.sh | bash -
# terrafom-docs
curl -Lo ./terraform-docs https://github.com/segmentio/terraform-docs/releases/download/v0.9.1/terraform-docs-v0.9.1-$(uname | tr '[:upper:]' '[:lower:]')-amd64
chmod +x ./terraform-docs
mv ./terraform-docs /usr/local/bin
# pre-commit
curl https://pre-commit.com/install-local.py | python -
```

---

It looks like it's failing because of the lack of a `.terraform-version`. We can fix that by just specifying the version. I don't like this though because now we're specifying it in two places
```suggestion
        tfenv install 0.12.25
```